lib/matplotlib/projections/polar.py

```diff
@@ -521,9 +523,17 @@ def set_thetagrids(self, angles, labels=None, frac=None, fmt=None,
         ACCEPTS: sequence of floats
         """
         # Make sure we take into account unitized data
-        angles = self.convert_yunits(angles)
-        angles = np.asarray(angles, np.float_)
-        self.set_xticks(angles * (np.pi / 180.0))
+        if isinstance( angles[0], float ):
+           # If the data is floats it is assumed to be degrees (as per the
+           # docsting for this method)
+           angles = np.asarray(angles, np.float_)
+           self.set_xticks(angles * (np.pi / 180.0))
+        else:
+           # The unit converters are defined to return radians
+           angles = self.convert_yunits(angles)
```

Comment:
Matplotlib's polar plots are `(theta, r)`; the `convert_[xy]units` calls assume `(r, theta)`.